static_frame/core/type_blocks.py

```diff
@@ -564,6 +564,20 @@ def mloc(self) -> np.ndarray:
     def unified(self) -> bool:
         return len(self._blocks) <= 1
 
+    @property
+    def unified_dtypes(self) -> bool:
```

Comment:
Wouldn't this be a good candidate for caching?

Reply:
Well, `TypeBlocks` are mutable, so caching would not be trivial. If we had a `TypeBlocks` and a `TypeBlocksGO`, that might make it easier.